### PR TITLE
stub: clean up call cancellation

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
Changes:

* No longer catch throwable, and don't wrap Error
* log an error if call.cancel() throws
* Unwrap ExecutionExcetion  always
* include a description with cancellation when interrupted

### CODE REVIEW

File: stub/src/main/java/io/grpc/stub/ClientCalls.java [hunk 1]
```diff
@@ -182,13 +190,17 @@ private ClientCalls() {}
    *     or an {@link InterruptedException}.
    */
   private static <V> V getUnchecked(Future<V> future) {
+
```
Comment (ejona86) [inline stub/src/main/java/io/grpc/stub/ClientCalls.java]: nit: Remove line addition?

File: stub/src/main/java/io/grpc/stub/ClientCalls.java [hunk 2]
```diff
@@ -214,6 +226,28 @@ private static StatusRuntimeException toStatusRuntimeException(Throwable t) {
...
+  /**
+   * Cancels a call, and throws the exception.
+   * 
+   * @param t must be a RuntimeException or Error
+   */
+  private static RuntimeException cancelThrow(ClientCall<?, ?> call, Throwable t) {
+    try {
+      call.cancel(null, t);
+    } catch (RuntimeException e) {
+      logger.log(Level.SEVERE, "RuntimeException encountered while closing call", e);
```
Comment (ejona86) [inline stub/src/main/java/io/grpc/stub/ClientCalls.java]: Why not just catch Throwable? These code paths are basically the same.

Comment (carl-mastrangelo) [reply stub/src/main/java/io/grpc/stub/ClientCalls.java]: I was hoping to (eventually) remove all the catch(throwable) in our code, and limit it only to the exceptions that can actually be thrown.  I can put it back the way it was if youd like, but it makes it harder to catch places where we do it wrong.  For example, we were wrapping Error above, previously.

Comment (ejona86) [reply stub/src/main/java/io/grpc/stub/ClientCalls.java]: Previously we weren't try-catching around cancel(). That honestly still sounds fine, but if you want the try-catch... sure. But I don't see any need for catching each type individually here.

File: stub/src/main/java/io/grpc/stub/ClientCalls.java (hunk 1 above)
Comment (carl-mastrangelo) [reply stub/src/main/java/io/grpc/stub/ClientCalls.java]: Done

File: stub/src/main/java/io/grpc/stub/ClientCalls.java (hunk 2 above)
Comment (carl-mastrangelo) [reply stub/src/main/java/io/grpc/stub/ClientCalls.java]: Done, but added an assert comment.

**Pull request merged**